gnovm/pkg/gnolang/values.go

```diff
@@ -725,6 +727,11 @@ func (mv *MapValue) GetLength() int {
 func (mv *MapValue) GetPointerForKey(alloc *Allocator, store Store, key *TypedValue) PointerValue {
 	kmk := key.ComputeMapKey(store, false)
 	if mli, ok := mv.vmap[kmk]; ok {
+		key2 := key.Copy(alloc)
+		// When assigning to a map item, the key is always equal to that of the
+		// last assignment; this is mostly noticeable in the case of -0 / 0:
+		// https://go.dev/play/p/iNPDR4FQlRv
+		mli.Key = key2
```

Comment:
i don't see why mli needs to be modified here.

Reply:
cross reference: https://github.com/gnolang/gno/pull/4327